drivers/SmartThings/matter-switch/src/test/test_matter_multi_button.lua

```diff
@@ -116,29 +116,42 @@ local function configure_buttons()
   test.socket.capability:__expect_send(mock_device:generate_test_message("button5", button_attr.pushed({state_change = false})))
 end
 
+-- All messages queued and expectations set are done before the driver is actually run
 local function test_init()
+  -- we dont want the integration test framework to generate init/doConfigure, we are doing that here
+  -- so we can set the proper expectations on those events.
+  test.disable_startup_messages()
+  test.mock_device.add_test_device(mock_device) -- make sure the cache is populated
+
+  -- added sets a bunch of fields on the device, and calls init
   local subscribe_request = CLUSTER_SUBSCRIBE_LIST[1]:subscribe(mock_device)
   for i, clus in ipairs(CLUSTER_SUBSCRIBE_LIST) do
     if i > 1 then subscribe_request:merge(clus:subscribe(mock_device)) end
   end
   test.socket.matter:__expect_send({mock_device.id, subscribe_request})
-  mock_device:set_field("__ELECTRICAL_TOPOLOGY", {topology = false, tags_on_ep = {}}, {persist = false}) -- since we're assuming this would have happened during device_added in this case.
```

Comment:
Note how this hack is no longer necessary because it is happening in the actual driver added/init/doConfigure handlers.